### PR TITLE
Backend/auth

### PR DESCRIPTION
**Pull request by s0lvang**
Backend for authentication through feide.

### CODE REVIEW

File: backend/auth.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    return decorated
+
+
+def requiresAdmin(f):
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        accessToken = get_token_auth_header("AccessToken")
+        groups = dataporten.getDataportenGroups(accessToken)
+        isAdmin = dataporten.checkIfAdmin(groups)
+        if(not isAdmin):
```
Comment (FluidSense) [inline backend/auth.py]: Unnecessary parenthesis?

File: backend/auth.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    """
+
+    @wraps(f)
+    @requiresIdToken()
+    def decorated(*args, **kwargs):
+        ctx = _request_ctx_stack.top
+        sub = ctx.idToken.get("sub")
+        user = userService.getUserFromSub(sub)
+        if(not user):
+            return Response(json.dumps({'error': 'User_not_exist'}), 401)
```
Comment (FluidSense) [inline backend/auth.py]: Does this need to be json-dumped? Can't it just be a text string?

File: backend/services/applicationService.py [hunk 3]
```diff
@@ -25,15 +25,16 @@ def getApplicationByUsername(username):
 
 
 # TODO application should not be registered with username, it should verify the user.
```
Comment (FluidSense) [inline backend/services/applicationService.py]: Don't it?

File: backend/controllers/userController.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+from flask import _request_ctx_stack
+from utils import dataporten
+
+user = Blueprint("user", __name__, url_prefix="/user")
+
+
+@user.route("/<id>")
+@requiresAdmin
+def getUser(id):
+    userApplication = userService.getUserById(id)
```
Comment (FluidSense) [inline backend/controllers/userController.py]: does `getUserById` exist on userService?

File: backend/models/user.py [hunk 5]
```diff
@@ -9,23 +9,37 @@ class User(db.Model):
...
+    )
+
     username = db.Column(
         "username",
         db.String(30),
         unique=True)
 
+    email = db.Column(
+        "email",
+        db.String(30),
```
Comment (FluidSense) [inline backend/models/user.py]: 30 is a bit short, isn't it?

File: backend/auth.py (hunk 2 above)
Comment (s0lvang) [reply backend/auth.py]: Also unnessecary parenthesis

**Pull request merged**